pkg/services/ngalert/remote/client/alertmanager_configuration.go

```diff
@@ -15,14 +16,25 @@ const (
 	grafanaAlertmanagerReceiversPath = "/api/v1/grafana/receivers"
 )
 
+type GrafanaAlertmanagerConfig struct {
+	TemplateFiles      map[string]string                    `yaml:"template_files" json:"template_files"`
+	AlertmanagerConfig definition.PostableApiAlertingConfig `yaml:"alertmanager_config" json:"alertmanager_config"`
+}
+
+func (u *GrafanaAlertmanagerConfig) MarshalJSON() ([]byte, error) {
+	// this is special marshaling that makes sure that secrets are not masked
```

Comment:
Nit:
```suggestion
	// This is special marshaling that makes sure that secrets are not masked.
```

Reply:
I will address it in https://github.com/grafana/grafana/pull/107808